File: frontends/frontend-qooxdoo/cs2-models/source/class/cs2/models/SSLIso3166.js
/*
    (DC)² - DataCenter Deployment Control
    Copyright (C) 2010  Stephan Adig <sh@sourcecode.de>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/

qx.Class.define("cs2.models.SSLIso3166",
{
  extend: dc2.models.DefaultModel,
  construct:function(RPCUrl) {
    this.base(arguments,RPCUrl);
    this._tableModel.setColumnIds(["_id","country_name","country_code"]);
    this._tableModel.setColumns(["ID","Country Name","ISO3166 Country Code"]);         
  },
  members: {
    getModelName:function() {
      return("cs2.models.SSLIso3166");
    },
    getModelCaption:function() {
      return("SSL ISO 3166 Country codes");
    },
    listData:function(search) {
      var _this=this;
      var handler=function(result,ex,id) {
        if (ex!=null) {
          // show error
        } else {
          _this._setTableData(result);
        }
      };
      if (search == null) {
        this._rpc.callAsync(handler,"cs2.ssl.iso3166.list");
      } else {
        // this._rpc.callAsync(handler,"cs2.ssl.keys.list",search);
      }
    },
    getIso3166List:function() {
      var result=this._rpc.callSync("cs2.ssl.iso3166.list");
      return(result);
    },
    addData:function(data) {
//      if (data != null) {        
//        try {
//          if (data["cipher"]=="None") {
//            data["cipher"]=null;
//          }
//          var response=this._rpc.callSync("cs2.ssl.keys.create",data["keyname"],data["description"],data["type"],data["bits"],data["cipher"],data["passphrase"]);
//          return(true);
//        } catch(exc) {
//          return(false);
//        }
//      }
    },
    deleteData:function(data) {
//      if (data != null) {
//        try {
//          var response=this._rpc.callSync("cs2.ssl.keys.remove",data["keyname"]);
//          return(true);
//        } catch(exc) {
//          return(false);
//        }        
//      }
    },
    getEmptyData:function() {
//      var data={};
//      data["_id"]=null;
//      data["keyname"]="New Keyname";
//      data["description"]="New description";
//      data["key_type"]="";
//      data["key_bits"]="";
//      data["cipher"]="";
//      data["passphrase"]="";
//      data["key_pem"]="";
//      return(data);
    }
  }
});

File: components/dc2-appserver/cs2/cs2rpcmethods/ssl_iso3166.py
# -*- coding: utf-8 -*-
#################################################################################
#
#    (DC)² - DataCenter Deployment Control
#    Copyright (C) 2010, 2011  Stephan Adig <sh@sourcecode.de>
#    This program is free software; you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation; either version 2 of the License, or
#    (at your option) any later version.
#
#    This program is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License along
#    with this program; if not, write to the Free Software Foundation, Inc.,
#    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#################################################################################
import types
import xmlrpclib
import re
from settings import MONGOS
from mongodb import Database, Table
import uuid
from helpers import check_record, datetime_isoformat
from rpc import rpcmethod
import web

tbl_iso3166 = Table(MONGOS["cs2broker"]["database"].get_table("iso3166"))

@rpcmethod(name="cs2.ssl.iso3166.list",returns={},params={},is_xmlrpc=True,is_jsonrpc=True)
def cs2_iso3166_list():
    isolist=tbl_iso3166.find()
    return isolist

@rpcmethod(name="cs2.ssl.iso3166.add",returns={},params={},is_xmlrpc=True,is_jsonrpc=True)
def cs2_iso3166_add(country_name=None,country_code=None):
    if country_name is None or country_name == "":
        return xmlrpclib.Fault(-32500, "Countryname can't be None or empty")
    if country_code is None or country_code == "":
        return xmlrpclib.Fault(-32500, "Country Code can't be None or empty")
    #
    # Check for duplicate
    #
    if tbl_iso3166.find_one({"country_code":country_code}) is not None:
        return xmlrpclib.Fault(-32500, "Country Code already exists in database")
    try:
        cdict={}
        cdict["country_name"]=country_name
        cdict["country_code"]=country_code
        tbl_iso3166.save(cdict)
        return cdict
    except Exception,e:
        return xmlrpclib.Fault(-32500, "Can't save to database")    
    return xmlrpclib.Fault(-32500, "Something really horrible happened")

    

